var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -219,32 +209,89 @@ def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
     def setup_dependent_package(self, module, dependent_spec):
         module.qmake = Executable(join_path(self.spec.prefix.bin, 'qmake'))
 
+    @when('@4 platform=darwin')
+    def patch(self):
+        ogl = self.spec['opengl'] if '+opengl' in self.spec else None
+        deployment_target = str(MACOS_VERSION.up_to(2))
+
+        patches = {
+            'MACOSX_DEPLOYMENT_TARGET': deployment_target,
+            'PREFIX': self.prefix,
+            'OPENGL_INCDIR': ogl.prefix.include if ogl else "",
+            'OPENGL_LIBS': ogl.libs.ld_flags if ogl else "",
+        }
+
+        def repl(match):
+            # Try to replace with our patch
```

Comment:
IMO a better wording: "replace the original config variable value with the one chosen here if it is mentioned in 'patches'; otherwise return the original value.

Reply:
Updated.